packages/liquidator/src/liquidator.js

```diff
@@ -250,52 +273,116 @@ class Liquidator {
-        from: this.account,
-        gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
-        gasPrice: this.gasEstimator.getCurrentFastPrice()
-      };
-      this.logger.debug({
-        at: "Liquidator",
-        message: "Liquidating position",
-        position: position,
-        inputPrice: scaledPrice.toString(),
-        minLiquidationPrice: this.liquidationMinPrice,
-        maxLiquidationPrice: maxCollateralPerToken.toString(),
-        tokensToLiquidate: tokensToLiquidate.toString(),
-        liquidationId: liquidationResult.liquidationId,
-        tokensLiquidated: liquidationResult.tokensLiquidated.rawValue.toString(),
-        finalFeeBond: liquidationResult.finalFeeBond.rawValue.toString(),
-        txnConfig
-      });
+        const reserveWeiNeededBN = this.toBN(reserveWeiNeeded);
+        const oneGweiReturnBN = this.toBN(oneGweiReturn);
+        const tokensToLiquidateBN = this.toBN(tokensToLiquidate);
+        const oneGweiBN = this.toBN(this.toWei("1", "gwei"));
+        const oneWeiBN = this.toBN(this.toWei("1"));
+
+        // Slippage = (idealReserveWeiNeeded - reserveWeiNeeded) / idealReserveWeiNeeded
+        // Where idealReserverWei = oneGweiReturn * tokensToLiquidate / 1 gwei
+        const idealReserveWeiNeededBN = oneGweiReturnBN.mul(tokensToLiquidateBN).div(oneGweiBN);
+
+        let slippage;
+        // IdealReserveWeiNeededBN is 0 if it isn't on exchanges yet
+        // Can't div by 0 so doing this
+        if (idealReserveWeiNeededBN.eq(this.toBN("0"))) {
+          slippage = "0";
+        } else {
+          slippage = this.fromWei(
+            idealReserveWeiNeededBN
+              .sub(reserveWeiNeededBN)
+              .mul(oneWeiBN)
+              .div(idealReserveWeiNeededBN)
+          );
+        }
+
+        // Don't allow slippage of more than 10%
+        if (parseFloat(slippage) > 0.1) {
+          this.logger.debug({
+            at: "Liquidator",
+            message: `Slippage too big while converting from reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${this.syntheticToken.address} ❌`,
+            oneGweiReturn,
+            reserveWeiNeeded,
+            idealReserveWeiNeeded,
+            slippage,
+            sponsor: position.sponsor,
+            inputPrice: scaledPrice.toString(),
+            position: position,
+            minLiquidationPrice: this.liquidationMinPrice,
+            maxLiquidationPrice: maxCollateralPerToken.toString(),
+            tokensToLiquidate: tokensToLiquidate.toString()
+          });
+          continue;
+        }
+
+        // Swap tokens
+        try {
+          await this.oneInchClient.swap(
+            {
+              fromToken: this.reserveCurrencyAddress,
+              toToken: this.syntheticToken.address,
+              minReturnAmountWei: tokensToLiquidate.toString(),
+              amountWei: reserveWeiNeededBN
+                .mul(this.toBN(this.toWei("1.005")))
```

Comment:
this is also a magic number and should be pulled into a constant. consider adding additional comments as well.